### PR TITLE
Chore/fix logo and optimization

### PR DESCRIPTION
**Pull request by thaihoa-fe**

### CODE REVIEW

File: src/components/TOC/withCategories.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+function useCategories(allMarkdownRemark) {
+    return useMemo(() => {
+        return allMarkdownRemark.edges
+            .reduce((result, edge) => {
+                return result.concat(normalizeHeading(edge.node));
+            }, []);
+    }, [allMarkdownRemark]);
+}
```
[the remark this answers is not in the document]
Comment (toannguyents) [reply src/components/TOC/withCategories.jsx]: - I suggest to remove HOC `withCategories` to make your code simpler. You can put those logic into TOC component directly or create a customHook if you intend to reuse the logic somewhere else

[the remark this answers is not in the document]
Comment (thaihoa-fe) [reply src/components/TOC/withCategories.jsx]: > * I suggest to remove HOC `withCategories` to make your code simpler. You can put those logic into TOC component directly or create a customHook if you intend to reuse the logic somewhere else

The reason to implement withCategories is to keep the TOC component pure, it's easier for testing.

File: src/components/TOC/withCategories.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return allMarkdownRemark.edges.reduce(
+      (result, edge) => [...result, ...normalizeHeading(edge.node)],
+      []
+    );
+  }, [allMarkdownRemark]);
+}
+
+export default function withCategories(WrapedComponent) {
+  return props => {
+    const { allMarkdownRemark } = useStaticQuery(graphql`
```
Comment (nhahoangts) [inline src/components/TOC/withCategories.js]: I think it's more clean to put `useStaticQuery` in `useCategories` custom hooks, then we don't have to pass `allMarkdownRemark` around.

Comment (thaihoa-fe) [reply src/components/TOC/withCategories.js]: ok I will fix in next commit

**Pull request merged**